db/migrate/20231006123736_add_unique_index_to_course_exports_cohort.rb

```diff
@@ -0,0 +1,48 @@
+class AddUniqueIndexToCourseExportsCohort < ActiveRecord::Migration[6.1]
+
+  class CourseExportsCohort < ApplicationRecord; end
+
+  def up
+    puts 'Finding duplicate records for CourseExportsCohort...'
+
+    duplicates = CourseExportsCohort.group(
+      :course_export_id,
+      :cohort_id
+    ).having('COUNT(*) > 1')
+
+    if duplicates.exists?
+      puts 'Found duplicates...'
+      puts 'Deleting the duplicates...'
+
+      duplicate_records = CourseExportsCohort.where(
+        course_export_id: duplicates.pluck(:course_export_id),
+        cohort_id: duplicates.pluck(:cohort_id)
+      )
+
+      duplicate_records.find_each do |duplicate|
+        puts "Deleting duplicate for course_exports_cohort.id: #{duplicate.id}..."
+        # Find and keep one of the duplicates
+        keep_one = CourseExportsCohort.find_by(
+          course_export_id: duplicate.course_export_id,
+          cohort_id: duplicate.cohort_id
+        )
+
+        # Find all duplicates with the same values and delete them
+        duplicates_to_delete = CourseExportsCohort.where(
+          course_export_id: duplicate.course_export_id,
+          cohort_id: duplicate.cohort_id
+        ).where.not(id: keep_one.id)
+
+        duplicates_to_delete.delete_all
+      end
+    end
```

Comment:
```suggestion
    puts 'Finding duplicate records for CourseExportsCohort...'

    CourseExportsCohort.select('MAX(created_at) as latest_time, course_export_id, cohort_id')
                       .group(:course_export_id, :cohort_id)
                       .having('COUNT(*) > 1')
                       .each do |duplicate|
      most_recent_record = CourseExportsCohort.where(course_export_id: duplicate.course_export_id,
                                                     cohort_id: duplicate.cohort_id)
                                              .order(created_at: :desc)
                                              .first

      CourseExportsCohort.where(course_export_id: duplicate.course_export_id,
                                cohort_id: duplicate.cohort_id)
                         .where.not(id: most_recent_record.id)
                         .delete_all
    end

```

Reply:
This can be made concise in one pass.

Reply:
Updated

Reply:
Hey @mahesh-krishnakumar  - We cannot use `created_at` here because it's not present in this table `course_exports_cohorts`

Reply:
Hey @mahesh-krishnakumar  - 

- We cannot use `created_at` here because it's not present in this table `course_exports_cohorts`.
- However, I've altered the logic a bit and replaced `created_at` with the `max_id`.
- I believe we can achieve the same thing with this. What do you say?